File: Go/bot.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"

	"github.com/bwmarrin/discordgo"
)

func main() {
	// Create a new Discord session using the provided bot token.
	dg, err := discordgo.New("Bot YOUR_BOT_TOKEN_HERE")
	if err != nil {
		fmt.Println("Error creating Discord session: ", err)
		return
	}

	// Register the messageCreate func as a callback for MessageCreate events.
	dg.AddHandler(messageCreate)

	// Open a websocket connection to Discord and begin listening.
	err = dg.Open()
	if err != nil {
		fmt.Println("Error opening Discord connection: ", err)
		return
	}

	// Wait here until CTRL-C or other term signal is received.
	fmt.Println("Bot is now running. Press CTRL-C to exit.")
	sc := make(chan os.Signal, 1)
	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt, os.Kill)
	<-sc

	// Cleanly close down the Discord session.
	dg.Close()
}

func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
	// Ignore messages sent by the bot itself.
	if m.Author.ID == s.State.User.ID {
		return
	}

	// If the message is "!hello", respond with "Hello, world!".
	if m.Content == "!hello" {
		s.ChannelMessageSend(m.ChannelID, "Hello, world!")
	}
}

File: C++/bot.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <discordcpp/restapi.hpp>
#include <discordcpp/gateway.hpp>
#include <discordcpp/client.hpp>

using namespace discordcpp;

int main(int argc, char* argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <bot token>" << std::endl;
        return 1;
    }

    std::string token = argv[1];
    client bot(token);

    bot.on_message([&bot](const message& msg) {
        if (msg.content == "!hello") {
            bot.create_message(msg.channel_id, "Hello, " + msg.author.username + "!");
        }
    });

    bot.run();

    return 0;
}
